go/vt/vtgate/planbuilder/operators/SQL_builder.go

```diff
@@ -312,6 +317,20 @@ func buildQuery(op ops.Operator, qb *queryBuilder) error {
 		for _, name := range op.Columns {
 			qb.addProjection(&sqlparser.AliasedExpr{Expr: name})
 		}
+	case *Projection:
+		err := buildQuery(op.Source, qb)
+		if err != nil {
+			return err
+		}
+
+		qb.clearProjections()
+		for i, column := range op.Columns {
+			ae := &sqlparser.AliasedExpr{Expr: column.GetExpr()}
+			if op.ColumnNames[i] != "" {
+				ae.As = sqlparser.NewIdentifierCI(op.ColumnNames[i])
+			}
+			qb.addProjection(ae)
+		}
```

Comment:
nit: we could also move this to its own little function - that would take us one step closer to having all query serving `switch` easily readable :)